src/Combinatorics/Graphs/functions.jl

```diff
@@ -169,9 +353,13 @@ function add_vertex!(g::Graph{T}) where {T <: Union{Directed, Undirected}}
     return n_vertices(g) - 1 == old_nvertices
 end
 
+function add_vertex!(mg::MixedGraph)
+  return add_vertex!(directed_component(mg)) && add_vertex!(undirected_component(mg))
```

Comment:
If only the second call fails here (adding a vertex to the undirected component), wouldn't this leave the mixed graph in an inconsistent state?